pkg/quota/v1/resources.go

```diff
@@ -199,15 +206,32 @@ func ContainsPrefix(prefixSet []string, item corev1.ResourceName) bool {
 	return false
 }
 
-// Intersection returns the intersection of both list of resources
+// Intersection returns the intersection of both list of resources, deduped and sorted
 func Intersection(a []corev1.ResourceName, b []corev1.ResourceName) []corev1.ResourceName {
-	setA := ToSet(a)
-	setB := ToSet(b)
-	setC := setA.Intersection(setB)
-	result := []corev1.ResourceName{}
-	for _, resourceName := range setC.List() {
-		result = append(result, corev1.ResourceName(resourceName))
+	result := make([]corev1.ResourceName, 0, len(a))
+	for _, item := range a {
+		if Contains(result, item) {
+			continue
+		}
+		if !Contains(b, item) {
+			continue
+		}
+		result = append(result, item)
 	}
+	sort.Slice(result, func(i, j int) bool { return result[i] < result[j] })
+	return result
+}
+
+// Difference returns the list of resources resulting from a-b, deduped and sorted
+func Difference(a []corev1.ResourceName, b []corev1.ResourceName) []corev1.ResourceName {
+	result := make([]corev1.ResourceName, 0, len(a))
+	for _, item := range a {
```

Comment:
Can we do this:
```
setA := ToSet(a)
for _, item := range b {
  if (!setA.Has(item)) {
    result = append(result, item)
  }
}
```
Similar operation for finding elements from a which are not in b

The complexity is O(M+N)